src/main/java/athleticli/data/Goal.java

```diff
@@ -0,0 +1,98 @@
+package athleticli.data;
+
+import java.time.DayOfWeek;
+import java.time.LocalDate;
+import java.time.temporal.TemporalAdjusters;
+
+/**
+ * Defines the basic fields and methods for a goal.
+ */
+public abstract class Goal {
+    /**
+     * Defines different types of goal periods.
+     */
+    public enum Period {
+        WEEKLY,
+        MONTHLY
+    }
```

Comment:
Nice use of enum! We could add DAILY in a later iteration (especially for diet goals it is more convenient to track and plan your daily habits)